# Hoja de trabajo 1

## Task 1 Análisis Teórico


### 1. Como director de un proyecto de conducción autónoma, debe dimensionar el hardware para un nuevo vehículo. El sistema utiliza 8 cámaras que capturan video a resolución 4K UHD (3840 x 2160).
Debido a la necesidad de alto rango dinámico (HDR), los sensores operan a 12 bits por píxel (Raw
Bayer Pattern) a 60 FPS.Métrica A: Enfocada puramente en el flujo vehicular.

a. Calcule el tamaño exacto de una sola imagen (frame) cruda en Megabytes (MB).

b. Calcule el ancho de banda necesario (en Gbps) para transmitir el flujo de las 8 cámaras al
procesador central sin compresión.

c. Si su procesador tiene una memoria RAM reservada de 16 GB exclusivamente para el buffer
de video, ¿cuántos segundos de historia puede almacenar antes de empezar a sobrescribir
datos?

d. Basado en su resultado, ¿es viable enviar estos datos "crudos" a la nube en tiempo real
usando 5G? Justifique

In [12]:
print("a. ")
res = 3840 * 2160
bpframe = res * 12
print(f"4K resolution is {res} pixels.")
print(f"MegaBytes por imagen= {((res * 12)/8)/1048576}")

a. 
4K resolution is 8294400 pixels.
MegaBytes por imagen= 11.865234375


## Megabytes por imagenes = 11.865234375

In [17]:
print("b. ")
anchoBtotal = bpframe * 60
print(f"ancho total de banda en Gb en 8 camaras = {(anchoBtotal/1000000000)*8}")



b. 
ancho total de banda en Gb en 8 camaras = 47.775744


## Datos por seg por 8 cams en Gb/s = 47.78Gb/s

In [22]:
ramMb = 16384
DpS8Cams = 712.2*8

print(f"c. {ramMb/DpS8Cams}" )


c. 2.875596742488065


## Puede almacenar aprox. 2.9 segundos de video crudo

No es viable transmitir en tiempo real a la nube el flujo de video crudo generado por las 8 cámaras utilizando 5G.

El sistema produce aproximadamente 45.6 Gbps de datos continuos, mientras que las redes 5G, incluso en condiciones ideales, ofrecen velocidades máximas teóricas del orden de 20 Gbps, y en escenarios reales de movilidad vehicular el ancho de banda efectivo suele estar entre 1 y 3 Gbps, con latencias variables y posibles pérdidas de conexión.

Por estas razones, el procesamiento debe realizarse localmente.

### Considere un píxel con valor de intensidad Iin=50 en una imagen estándar de 8 bits (0−255). Se aplican dos procesos de mejora secuenciales en el siguiente orden:
### I. Corrección Gamma con γ=0.5 (para expandir sombras).
### II. Ajuste Lineal con ganancia α=1.2 y brillo β=-10 (para contrastar).

Realice los cálculos en el dominio de flotantes normalizados [0,1] como dicta la buena práctica y
convierta a entero de 8 bits solo al final.

a. Calcule el valor final del píxel I_out.

b. ¿Hubo saturación (clipping) en el proceso?

c. Si hubiéramos realizado las operaciones usando uint8 directamente sin convertir a float
(truncando decimales en cada paso intermedio), ¿cuál habría sido el error numérico
resultante?

In [ ]:
I = 50
Ii = 50/255
gamma = 0.5
alfa = 1.2
beta = -10

Ic = Ii**gamma
print(f"correccion gamma: {Ic}")

Bnorm = beta/255

Isalida = alfa * Ic + Bnorm
print(f"imagen de salida: {Isalida*255}")




correccion gamma: 0.44280744277004763
imagen de salida: 125.49907748763458


### a. El valor final del pixel es de aprox. 126

### b. No hubo stranding, porque el valor intermedio siempre estuvo dentro de 0 y 1 y el final dentro de 255

### c. El resultado final hubiera sido el 125 o sea 1 valor de intensidad

## Task 2 Práctica

## Task 3 Preguntas Post-Práctica

